prime-angular-frontend/src/app/modules/enrolment/pages/care-setting/care-setting.component.ts

```diff
@@ -94,6 +94,11 @@ export class CareSettingComponent extends BaseEnrolmentProfilePage implements On
       this.setHealthAuthorityValidator();
     }
 
+    // Remove device provider identifier if Device Provider is no longer selected
+    if (!controls.some(c => c.value.careSettingCode === CareSettingEnum.DEVICE_PROVIDER)) {
+      this.enrolmentFormStateService.regulatoryFormState.deviceProviderIdentifier.setValue(null);
```

Comment:
You can use `.reset()` it sets the value to null without you having to use `.setValue(null)`

Reply:
using reset now